### PR TITLE
feat: added a base where clause to Query class

### PR DESCRIPTION
**Pull request by mpskowron**
I've added a few questions as PR comments.

### CODE REVIEW

File: src/test/scala/sss/db/DbV2Spec.scala [hunk 1]
```diff
@@ -240,6 +215,16 @@ class DbV2Spec extends DbSpecSetup {
     } finally fixture.dbUnderTest.dropView("testview2").runSyncAndGet
   }
 
+  it should "support select query with non empty base where statement" in {
```
Comment (mpskowron) [inline src/test/scala/sss/db/DbV2Spec.scala]: Is this test enough or should I add more? If so,  should I create a separate QuerySpec?

File: src/main/scala/sss/db/package.scala [hunk 2]
```diff
@@ -228,7 +228,15 @@ package object db extends Logging {
...
-      copy(clause = newClause, params = params ++ w.params)
+      val newLimit = w.orderBys.limit.orElse(orderBys.limit)
+      copy(
+        clause = newClause,
+        params = params ++ w.params,
+        orderBys = orderBys.copy(
+          orderBys = orderBys.orderBys ++ w.orderBys.orderBys,
+          limit = newLimit
+        )
+      )
```
Comment (mpskowron) [inline src/main/scala/sss/db/package.scala:237]: This was changed, because and() didn't compose well with OrderBy.

**Review by mcsherrylabs — CHANGES_REQUESTED** [review]
Nice! 
I think we can add the where to the View constructor also and a test that exercises that.
And I didn't notice a test for the bug fix in `and()`? 
This is an improvement I believe. ta.

File: src/test/scala/sss/db/DbV2Spec.scala (hunk 1 above)
Comment (mcsherrylabs) [reply src/test/scala/sss/db/DbV2Spec.scala]: This is reasonable. Ive asked to add the baseWhere to View also. So if maybe QuerySpec? Up to you.

File: src/test/scala/sss/db/DbSpecSetup.scala [hunk 3]
```diff
@@ -21,19 +23,42 @@ trait DbSpecSetupBase extends Matchers with BeforeAndAfter {
...
+    finally fixture.dbUnderTest.shutdown.runSyncAndGet
   }
 
+  def persistIntVal(intVal: Int): FutureTx[Row] = fixture.table.persist(Map("strId" -> "strId", "createTime" -> new Date(), "intVal" -> intVal))
+  def persistIntVals(intVals: Seq[Int]): FutureTx[Seq[Row]] = FutureTx.sequence(intVals map persistIntVal)
 }
 
 trait DbSpecSetup extends AnyFlatSpec with DbSpecSetupBase
+
+trait DbSpecQuickSetupBase extends DbSpecSetupBase {
```
Comment (mpskowron) [inline src/test/scala/sss/db/DbSpecSetup.scala:45]: Added that to reduce boilerplate when using DbSpecSetup. Haven't changed DbSpecSetup itself, to avoid making a lot of changes at once. I didn't have a good naming idea, thus  the "Quick" in the middle.

File: src/main/scala/sss/db/Db.scala [hunk 4]
```diff
@@ -68,14 +66,17 @@ class Db(dbConfig: DbConfig)(closeableDataSource:CloseableDataSource, ec: Execut
 
   TL;DR for blockchain type applications views are a good solution.
    */
-  def view(name: String): View = new View(name, syncRunContext, dbConfig.freeBlobsEarly)
+  def view(name: String): View = viewWhere(name, where())
```
Comment (mcsherrylabs) [inline src/main/scala/sss/db/Db.scala]: Sorry I didn't spot this yesterday, but I think we can use a default param rather than selectWhere and viewWhere. 
`where` is a reasonable param to `select`. Slightly suprising as a param to `view` to it makes sense on inspection.

File: src/main/scala/sss/db/Query.scala [hunk 5]
```diff
@@ -190,21 +191,13 @@ class Query private[db] (private val selectSql: String,
...
-      val st = context.conn.createStatement()
-      try {
-        val clause = where() orderBy (orderClauses: _*) limit(start, pageSize)
-        val rs = st.executeQuery(s"${selectSql} ${clause.sql}")
-        Rows(rs, freeBlobsEarly)
-      } finally st.close()
-    }(context.ec)
-  }
+  def page(start: Long, pageSize: Int, orderClauses: Seq[OrderBy] = Seq(OrderAsc("id"))): FutureTx[Rows] =
+    filter(where() orderBy (orderClauses: _*) limit(start, pageSize))
```
Comment (mcsherrylabs) [inline src/main/scala/sss/db/Query.scala:195]: Nice. :+1:

File: src/test/scala/sss/db/DbSpecSetup.scala (hunk 3 above)
Comment (mcsherrylabs) [reply src/test/scala/sss/db/DbSpecSetup.scala:45]: I see a db shutdown after each .... I have a nagging memory of something going wrong with that... please try to run the testsuite repeatedly locally watching for errors. Strange request admittedly.

**Review by mcsherrylabs — COMMENTED** [review]
ok. One small change to reduce the API "surface" and a little testing and it's done!
